apps/cli/src/logger.ts

```diff
@@ -0,0 +1,274 @@
+interface LogChunk {
+  event: string;
+  data: any;
+  ops?: Array<{ value: string }>;
+}
+
+interface MessageData {
+  id: string;
+  type: "ai" | "tool" | "human";
+  content: Array<{ type: string; text?: string; input?: string }> | string;
+  tool_calls?: Array<ToolCall>;
+  name?: string;
+  tool_call_id?: string;
+  status?: "success" | "error";
+  graph_id?: string;
+  langgraph_node?: string;
+  additional_kwargs?: {
+    reasoning?: string;
+    notes?: string[];
+  };
+}
+
+interface ToolCall {
+  type: string;
+  args?: string;
+  name?: string;
+  index?: number;
+  id?: string;
+}
+/**
+ * Format a tool result based on its type and content
+ */
+function formatToolResult(message: MessageData): string {
+  const content =
+    typeof message.content === "string"
+      ? message.content
+      : message.content?.[0]?.text || message.content?.[0]?.input || "";
```

Comment:
after converting to a LC message you can do this:
```suggestion
  const content = getMessageContentString(message.content);
```
import `getMessageContentString` from shared/messages